File: Csharp/Lab8/Lab8/Lab8/cng/Model/Collections/Comparators/SizeComparatorToGreater.cs
using System.Collections;
using Lab8.cng.Model.Entity.Abstracts;

namespace Lab8.cng.Model.Collections.Comparators
{
    public class SizeComparatorToGreater : IComparer
    {
        public int Compare(object x, object y)
        {
            var t = (Toy) x;
            var t1 = (Toy) y;
            return (int) (t.GetSize() - t1.GetSize());
        }
    }
}

File: Kotlin/Lab8/src/by/bntu/fitr/povt/cng/model/collections/comparators/SizeComparatorToGreater.java
package by.bntu.fitr.povt.cng.model.collections.comparators;

import by.bntu.fitr.povt.cng.model.entity.abstracts.Toy;

import java.util.Comparator;

public class SizeComparatorToGreater implements Comparator<Toy> {
    @Override
    public int compare(Toy o1, Toy o2) {
        return (int) (o1.getSize() - o2.getSize());
    }
}
